python/GafferSceneUI/LightEditor.py

```diff
@@ -425,6 +431,23 @@ def __selectLinked (self, *unused ) :
 		if not isinstance( result, Exception ) :
 			GafferSceneUI.ContextAlgo.setSelectedPaths( context, result )
 
+	def __deleteLights( self, *unused ) :
+
+		# \todo What about :
+		# when editScope == None (no editScope selected)
+		# any similar case to EditScopeUI where it tests for a disabled edit scope?
+		# check that edit scope is downstream of focus node?
+		# read-only edit scopes?
+		# EditScopeAlgo.prunedReadOnly()?
+		# editScope enabled but pruningProcessor disabled?
```

Comment:
> when editScope == None (no editScope selected)

Grey out the menu item, so we avoid the error that occurs when you select it.

> check that edit scope is downstream of focus node?

That does seem worthwhile for something as drastic as deletion, since we already error for attempts to edit parameters in that state. I guess the problem is that the name column isn't an InspectorColumn, so doesn't provide the query we need? 

> EditScopeAlgo.prunedReadOnly()?

Grey out the menu item.

> any similar case to EditScopeUI where it tests for a disabled edit scope?
> editScope enabled but pruningProcessor disabled?

I think we should deal with these cases too, so people don't delete things while the processor has no effect, only to be surprised later when they turn it back on. We can use the less-than-ideal method from EditScopeUI for now...

Reply:
I believe I have all of those handled in 66f23915ac7c889f901e7b526413234db8a794a1 now.